src/reporting/Market.sol

```diff
@@ -0,0 +1,653 @@
+        // CONSIDER: should we allow creator to send extra ETH, is there risk of variability in bond requirements?
+        require(msg.value == MarketFeeCalculator(controller.lookup("MarketFeeCalculator")).getMarketCreationCost(_reportingWindow));
+        reportingWindow = _reportingWindow;
+        endTime = _endTime;
+        numOutcomes = _numOutcomes;
+        payoutDenominator = _payoutDenominator;
+        // FIXME: markets may be denominated in tokens that aren't 10^18, deal with that
+        feePerEthInAttoeth = _feePerEthInAttoeth;
+        maxDisplayPrice = _maxDisplayPrice;
+        minDisplayPrice = _minDisplayPrice;
+        marketCreationBlock = block.number;
+        topic = _topic;
+        automatedReporterAddress = _automatedReporterAddress;
+        denominationToken = _denominationToken;
+        creator = _creator;
+        for (uint8 _outcome = 0; _outcome < numOutcomes; _outcome++) {
+            shareTokens.push(createShareToken(_outcome));
+        }
+        approveSpenders();
+        require(controller.lookup("Cash") == address(denominationToken) || getBranch().isContainerForShareToken(denominationToken));
+        _success = true;
+        return _success;
+    }
+
+    function createShareToken(uint8 _outcome) private returns (IShareToken) {
+        return ShareTokenFactory(controller.lookup("ShareTokenFactory")).createShareToken(controller, this, _outcome);
+    }
+
+        // TODO: we need to update this signature (and all of the places that call it) to allow the creator (UI) to pass in a number of other things which will all be logged here
+        // TODO: log short description
+        // TODO: log long description
+        // TODO: log min display price
+        // TODO: log max display price
+        // TODO: log tags (0-2)
+        // TODO: log outcome labels (same number as numOutcomes)
+        // TODO: log type (scalar, binary, categorical)
+        // TODO: log any immutable data associated with the market (e.g., endTime, numOutcomes, payoutDenominator, denominationToken address, etc.)
+
+    // this will need to be called manually for each open market if a spender contract is updated
+    function approveSpenders() private returns (bool) {
+        bytes32[5] memory _names = [bytes32("cancelOrder"), bytes32("completeSets"), bytes32("takeOrder"), bytes32("tradingEscapeHatch"), bytes32("claimProceeds")];
+        uint8 i = 0;
+        for (i = 0; i < 5; i++) {
+            denominationToken.approve(controller.lookup(_names[i]), APPROVAL_AMOUNT);
+        }
+        for (i = 0; i < numOutcomes; i++) {
+            shareTokens[i].approve(controller.lookup("takeOrder"), APPROVAL_AMOUNT);
+        }
+        return true;
+    }
+
+    function changeCreator(address _newCreator) public returns (bool) {
+        require(msg.sender == creator);
+        creator = _newCreator;
+        return true;
+    }
```

Comment:
I wonder if we should change this contract to inherit from `Ownable` and change creator to owner everywhere in here.  This would allow us to use the `onlyOwner` modifier on this method and the other one(s) that need access control, and it would help get a little bit of code out of this contract (I recognize this won't help with contract size, but it will help with readability).

Note: Such a change would also require the factory to change the owner as part of contract initialization.